ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/RewrittenAnalysis.java

```diff
@@ -113,7 +123,49 @@ public ImmutableAnalysis original() {
 
   @Override
   public Optional<WindowExpression> getWindowExpression() {
-    return original.getWindowExpression();
+    if (original.getRefinementInfo().isPresent()
+        && original.getRefinementInfo().get().getOutputRefinement() == OutputRefinement.FINAL
+        && !original.getWindowExpression().get()
```

Comment:
nit: we're not checking `isPresent` here, which I think is causing https://jenkins.confluent.io/job/confluentinc-pr/job/ksql/job/PR-5874/1/testReport/junit/io.confluent.ksql.planner/LogicalPlannerTest/shouldThrowOnNonWindowedAggregationSuppressions/ to fail